internal/cmd/deploy.go

```diff
@@ -90,6 +96,14 @@ func (c *deployCommand) preRun(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("host must be set when using TLS")
 	}
 
+	if cmd.Flags().Changed("tls-certificate-path") && !cmd.Flags().Changed("tls-private-key-path") {
+		return fmt.Errorf("tls-private-key-path must be set when specified tls-certificate-path")
+	}
+
+	if cmd.Flags().Changed("tls-private-key-path") && !cmd.Flags().Changed("tls-certificate-path") {
+		return fmt.Errorf("tls-certificate-path must be set when specified tls-private-key-path")
+	}
```

Comment:
Could use [MarkFlagsRequiredTogether](https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsRequiredTogether) here instead of checking them manually, as it's a case that Cobra can handle. (Similar to how we're using `MarkFlagsOneRequired` [here](https://github.com/basecamp/kamal-proxy/blob/main/internal/cmd/rollout_set.go#L28)).

Reply:
I initially tried `MarkFlagsRequiredTogether`, but the error was looking pretty weird:

> Error: if any flags in the group [tls-certificate-path tls-private-key-path] are set they must all be set; missing [tls-private-key-path]

Question of usability?

Reply:
Updated the branch